builtin/providers/aws/resource_aws_db_snapshot.go

```diff
@@ -0,0 +1,215 @@
+		DBInstanceIdentifier: aws.String(d.Get("db_instance_identifier").(string)),
+		DBSnapshotIdentifier: aws.String(d.Get("db_snapshot_identifier").(string)),
+	}
+
+	_, err := conn.CreateDBSnapshot(params)
+	if err != nil {
+		return err
+	}
+	d.SetId(d.Get("db_snapshot_identifier").(string))
+
+	_, err = resourceAwsDbSnapshotWaitForAvailable(d.Id(), conn)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsDbSnapshotRead(d, meta)
+}
+
+func resourceAwsDbSnapshotRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	params := &rds.DescribeDBSnapshotsInput{
+		DBSnapshotIdentifier: aws.String(d.Id()),
+	}
+	resp, err := conn.DescribeDBSnapshots(params)
+	if err != nil {
+		return err
+	}
+
+	snapshot := resp.DBSnapshots[0]
+
+	d.Set("allocated_storage", snapshot.AllocatedStorage)
+	d.Set("availability_zone", snapshot.AvailabilityZone)
+	d.Set("db_snapshot_arn", snapshot.DBSnapshotArn)
+	d.Set("encrypted", snapshot.Encrypted)
+	d.Set("engine", snapshot.Engine)
+	d.Set("engine_version", snapshot.EngineVersion)
+	d.Set("iops", snapshot.Iops)
+	d.Set("kms_key_id", snapshot.KmsKeyId)
+	d.Set("license_model", snapshot.LicenseModel)
+	d.Set("option_group_name", snapshot.OptionGroupName)
+	d.Set("port", snapshot.Port)
+	d.Set("source_db_snapshot_identifier", snapshot.SourceDBSnapshotIdentifier)
+	d.Set("source_region", snapshot.SourceRegion)
+	d.Set("snapshot_type", snapshot.SnapshotType)
+	d.Set("status", snapshot.Status)
+	d.Set("vpc_id", snapshot.VpcId)
+
+	return nil
+}
+
+func resourceAwsDbSnapshotDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	params := &rds.DeleteDBSnapshotInput{
+		DBSnapshotIdentifier: aws.String(d.Id()),
+	}
+	_, err := conn.DeleteDBSnapshot(params)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsDbSnapshotWaitForAvailable(id string, conn *rds.RDS) (*rds.DBSnapshot, error) {
+	log.Printf("Waiting for Snapshot %s to become available...", id)
+
+	req := &rds.DescribeDBSnapshotsInput{
+		DBSnapshotIdentifier: aws.String(id),
+	}
+	pollsWhereNotFound := 0
+	for {
+		res, err := conn.DescribeDBSnapshots(req)
+		if err != nil {
+			if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "DBSnapshotNotFound" {
+				pollsWhereNotFound++
+				// We arbitrarily stop polling after getting a "not found" error five times,
+				// assuming that the Snapshot has been deleted by something other than Terraform.
+				if pollsWhereNotFound > 100 {
+					return nil, fmt.Errorf("gave up waiting for Snapshot to be created: %s", err)
+				}
+				time.Sleep(30 * time.Second)
+				continue
+			}
+			return nil, fmt.Errorf("error reading DB Snapshot: %s", err)
+		}
+
+		if len(res.DBSnapshots) != 1 {
+			return nil, fmt.Errorf("new Snapshot vanished while pending")
+		}
+
+		state := *res.DBSnapshots[0].Status
+
+		if state == "creating" {
+			// Give it a few seconds before we poll again.
+			time.Sleep(4 * time.Second)
```

Comment:
So we don't actually time out on snapshot creation, we poll indefinitely until it completes?  That's probably the right thing, considering this is _data_, but I wonder if some users would like to toggle this behavior, e.g. "create a snapshot, but I trust that AWS will complete it and I don't want/need Terraform to sit and wait for it to finish". Not a blocker, but a thought